### PR TITLE
Start of upgrade

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
 projectName: clowder
 repo: cloud.redhat.com/clowder/v2
 resources:
-- group: cloud.redhat.com
-  kind: ClowdApp
-  version: v1alpha1
-- group: cloud.redhat.com
+- api:
+    crdVersion: v1
+    namespaced: true
```
Comment (kylape) [inline PROJECT:13]: ClowdEnvironments aren't namespaced

File: apis/cloud.redhat.com/v1alpha1/clowdapp_webhook.go [hunk 2]
```diff
@@ -35,6 +35,11 @@ func (r *ClowdApp) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (kylape) [inline apis/cloud.redhat.com/v1alpha1/clowdapp_webhook.go:38]: can we remove this comment? I think it's pretty obvious that we should edit this file.

File: config/manager/controller_manager_config.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+kind: ControllerManagerConfig
```
Comment (kylape) [inline config/manager/controller_manager_config.yaml:2]: so now our manager pod is configured using this CR?

File: config/manifests/kustomization.yaml [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+- bases/clowder.clusterserviceversion.yaml
 - ../default
 - ../samples
 - ../scorecard
+
+# [WEBHOOK] To enable webhooks, uncomment all the sections with [WEBHOOK] prefix.
+# Do NOT uncomment sections with prefix [CERTMANAGER], as OLM does not support cert-manager.
+# These patches remove the unnecessary "cert" volume and its manager container volumeMount.
+#patchesJson6902:
+#- target:
```
Comment (kylape) [inline config/manifests/kustomization.yaml:13]: should we uncomment this?

File: config/rbac/auth_proxy_role_binding.yaml [hunk 5]
```diff
@@ -8,5 +8,5 @@ roleRef:
   name: proxy-role
 subjects:
 - kind: ServiceAccount
-  name: default
+  name: controller-manager
```
Comment (kylape) [inline config/rbac/auth_proxy_role_binding.yaml:11]: I don't think we use this role binding, but does the manager pod run under a different service account now?

File: config/webhook/kustomization.yaml [hunk 6]
```diff
@@ -1,7 +1,6 @@
 resources:
 - manifests.yaml
 - service.yaml
-- network_policy.yaml
```
Comment (kylape) [inline config/webhook/kustomization.yaml:4]: do we not need this anymore?

File: controllers/cloud.redhat.com/clowdapp_controller.go [hunk 7]
```diff
@@ -95,10 +95,10 @@ type ClowdAppReconciler struct {
 // +kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheuses;servicemonitors,verbs=get;list;watch;create;update;patch;delete
 
 // Reconcile fn
-func (r *ClowdAppReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+func (r *ClowdAppReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	qualifiedName := fmt.Sprintf("%s:%s", req.Namespace, req.Name)
 	log := r.Log.WithValues("app", qualifiedName).WithValues("id", utils.RandString(5))
-	ctx := context.WithValue(context.Background(), errors.ClowdKey("log"), &log)
+	ctx = context.WithValue(ctx, errors.ClowdKey("log"), &log)
```
Comment (kylape) [inline controllers/cloud.redhat.com/clowdapp_controller.go:101]: do we still need to add the logger to the context? I thought they said they added it in the controller-runtime

File: main.go [hunk 8]
```diff
@@ -18,47 +18,55 @@ package main
...
-
-	if clowder_config.LoadedConfig.DebugOptions.Pprof.CpuFile != "" {
-		f, err := os.Create(clowder_config.LoadedConfig.DebugOptions.Pprof.CpuFile)
-		if err != nil {
-			log.Fatal(err)
-		}
-		pprof.StartCPUProfile(f)
-		defer pprof.StopCPUProfile()
+	opts := zap.Options{
+		Development: true,
```
Comment (kylape) [inline main.go:61]: did we flip devmode from false to true in this change?

File: config/manager/controller_manager_config.yaml (hunk 3 above)
Comment (psav) [reply config/manager/controller_manager_config.yaml:2]: yes

File: config/manifests/kustomization.yaml (hunk 4 above)
Comment (psav) [reply config/manifests/kustomization.yaml:13]: I'm really not sure

File: config/rbac/auth_proxy_role_binding.yaml (hunk 5 above)
Comment (psav) [reply config/rbac/auth_proxy_role_binding.yaml:11]: yes I think so

File: config/webhook/kustomization.yaml (hunk 6 above)
Comment (psav) [reply config/webhook/kustomization.yaml:4]: Added back in - we may need to tweak this to the version you created - this was not generated by operator-sdk

File: controllers/cloud.redhat.com/clowdapp_controller.go (hunk 7 above)
Comment (psav) [reply controllers/cloud.redhat.com/clowdapp_controller.go:101]: I'm not sure - but this seems to work as expected

File: main.go (hunk 8 above)
Comment (psav) [reply main.go:61]: good catch

File: config/manifests/kustomization.yaml (hunk 4 above)
Comment (kylape) [reply config/manifests/kustomization.yaml:13]: works in minikube, will see if it works in olm

**Review by kylape — APPROVED** [review]

**Comment by psav** [comment]
This was merged in another branch

**Pull request closed without merging**